plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClusteringPredicatesExtractor.java

```diff
@@ -65,102 +67,135 @@ private static ClusteringPushDownResult getClusteringKeysSet(List<CassandraColum
-                            else {
-                                predicate = CassandraCqlUtils.validColumnName(columnHandle.getName()) + " IN ("
-                                        + Joiner.on(",").join(singleValues) + ")";
-                            }
-                        }
-                        else if (!rangeConjuncts.isEmpty()) {
-                            predicate = Joiner.on(" AND ").join(rangeConjuncts);
-                        }
-                        return predicate;
-                    }, discreteValues -> {
-                        if (discreteValues.isInclusive()) {
-                            ImmutableList.Builder<Object> discreteValuesList = ImmutableList.builder();
-                            for (Object discreteValue : discreteValues.getValues()) {
-                                discreteValuesList.add(columnHandle.getCassandraType().toCqlLiteral(discreteValue));
-                            }
-                            String predicate = CassandraCqlUtils.validColumnName(columnHandle.getName()) + " IN ("
-                                    + Joiner.on(",").join(discreteValuesList.build()) + ")";
-                            return predicate;
+                            String inValues = discreteValues.getValues().stream()
+                                    .map(columnHandle.getCassandraType()::toCqlLiteral)
+                                    .collect(joining(","));
+                            fullyPushedDomains.add(columnHandle);
+                            return CassandraCqlUtils.validColumnName(columnHandle.getName()) + " IN (" + inValues + " )";
                         }
                         return null;
                     }, allOrNone -> null);
 
             if (predicateString == null) {
                 break;
             }
-            // IN restriction only on last clustering column for Cassandra version = 2.1
-            if (predicateString.contains(" IN (") && cassandraVersion.compareTo(VersionNumber.parse("2.2.0")) < 0 && currentClusteringColumn != (clusteringColumns.size() - 1)) {
-                break;
-            }
             clusteringColumnSql.add(predicateString);
-            domainsBuilder.put(columnHandle, domain);
             // Check for last clustering column should only be restricted by range condition
             if (predicateString.contains(">") || predicateString.contains("<")) {
                 break;
             }
-            currentClusteringColumn++;
+            allProcesedClusteringColumns++;
         }
         List<String> clusteringColumnPredicates = clusteringColumnSql.build();
 
-        return new ClusteringPushDownResult(domainsBuilder.build(), Joiner.on(" AND ").join(clusteringColumnPredicates));
+        return new ClusteringPushDownResult(fullyPushedDomains.build(), Joiner.on(" AND ").join(clusteringColumnPredicates));
+    }
+
+    /**
+     * IN restriction allowed only on last clustering column for Cassandra version <= 2.1
```

Comment:
nit: Adjust the comment since the condition checks for v < 2.2.0 (there are versions between 2.1 and 2.2.0).